components/financial_assistance/app/assets/javascripts/financial_assistance/income.js

```diff
@@ -1,33 +1,42 @@
   // enable inputs on drivers, instructions, and income kinds - except for submit (save) buttons as this would enable the submit button in the dummy forms, and fake-disabled-inputs as they should never be enabled
   $('.driver-question input, .instruction-row input, .income input:not(":input[type=submit], .fake-disabled-input"), .other-income-kind input:not(":input[type=submit], .fake-disabled-input"), .unemployment-income input:not(":input[type=submit], .fake-disabled-input")').removeAttr('disabled');
   
-    // enable "Add New" income buttons
-  $('a.new-income').removeClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').removeAttr('disabled');
+  // enable "Add New" income buttons
+  $('#new-unemployment-income, #new-ai-an-income, .new-income, .add_new_other_income_kind').removeAttr('disabled');
   $("a.interaction-click-control-add-more").removeClass('hide');
   $("a[class*='income-edit'], a[class*='income-delete']").removeClass('disabled'); // legacy
+  $('#self-employed-income').removeClass('disabled');
+  $('a.new-income').removeClass('disabled');
+  $('a.self-employed-income-delete').removeClass('disabled');
 
   // enable nav
   $('#nav-buttons a').removeClass('disabled');
   $('.col-md-3 > .interaction-click-control-continue').removeClass('disabled'); // legacy
+
 };
 
 function startEditingIncome(income_kind) {
+  shouldShowEndDateWarningModal = true;
   // apply disabled style to drivers, instructions, and other income kinds not being edited
   $('.driver-question, .driver-question a, .instruction-row, .income:not(#' + income_kind + '), .other-income-kind:not(#' + income_kind + '), .unemployment-income:not(#' + income_kind + ')').addClass('disabled');
   // disable inputs on drivers, instructions, and other income kinds not being edited - except for submit (save) buttons as some form types require them to be enabled on edit/create
   $('.driver-question input, .instruction-row input, .income:not(#' + income_kind + ') input:not(":input[type=submit]"), .other-income-kind:not(#' + income_kind + ') input:not(":input[type=submit]"), .unemployment-income:not(#' + income_kind + ') input:not(":input[type=submit]")').attr('disabled', true);
-  
+    
   // disable "Add New" income buttons
-  $('a.new-income').addClass('hide');
-  $('#new-unemployment-income, #new-ai-an-income, .add_new_other_income_kind').attr('disabled', true);
+  $('#self-employed-income').addClass('disabled');
+  $('a.new-income').addClass('disabled');
+  $('#new-unemployment-income, #new-ai-an-income, .new-income, .add_new_other_income_kind').attr('disabled', true);
   $("a.interaction-click-control-add-more").addClass('hide'); // legacy
 
   // disable all income edit and edit buttons on created incomes
   $("a[class*='income-edit'], a[class*='income-delete']").addClass('disabled');
+  $('a.income-edit').addClass('disabled');
```

Comment:
the above selector `"a[class*='income-edit'], a[class*='income-delete']` should already match these two you added. do you need these two?